### PR TITLE
Added page-number argument to website-article-controller

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | none
| Related issues/PRs | build ontop #122
| License | MIT
| Documentation PR | none

#### What's in this PR?

This PR populates the article-page to the website.

#### To Do

- [x] Tests

### CODE REVIEW

File: Routing/ArticleRouteDefaultProvider.php [hunk 1]
```diff
@@ -66,11 +67,17 @@ public function getByEntity($entityClass, $id, $locale, $object = null)
             $object = $this->documentManager->find($id, $locale);
         }
 
+        $pageNumber = $object->getPageNumber();
+        if ($object instanceof ArticlePageDocument) {
+            $object = $object->getParent();
```
Comment (wachterjohannes) [inline Routing/ArticleRouteDefaultProvider.php]: always article is used for rendering (serialization handles page-number).

File: Controller/WebsiteArticleController.php [hunk 2]
```diff
@@ -28,11 +28,12 @@ class WebsiteArticleController extends Controller
...
      * @param ArticleInterface $object
+     * @param int $pageNumber
      * @param string $view
      *
      * @return Response
      */
-    public function indexAction(Request $request, ArticleInterface $object, $view)
+    public function indexAction(Request $request, ArticleInterface $object, $pageNumber, $view)
     {
         $content = $this->get('jms_serializer')->serialize(
```
Comment (alexander-schranz) [inline Controller/WebsiteArticleController.php]: make function for this so when I overwrite the WebsiteArticleCOntroller I dont need to remember this.

File: Controller/WebsiteArticleController.php [hunk 3]
```diff
@@ -28,11 +28,12 @@ class WebsiteArticleController extends Controller
      *
      * @param Request $request
      * @param ArticleInterface $object
+     * @param int $pageNumber
      * @param string $view
      *
      * @return Response
      */
-    public function indexAction(Request $request, ArticleInterface $object, $view)
+    public function indexAction(Request $request, ArticleInterface $object, $pageNumber, $view)
```
Comment (alexander-schranz) [inline Controller/WebsiteArticleController.php]:
```
indexAction(Request $request, ArticleInterface $object, $view, $pageNumber) // or $pageNumber = 1
```

File: Document/Serializer/ArticleWebsiteSubscriber.php [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+     * @return array
+     */
+    private function resolve(ArticleInterface $article)
+    {
+        $structure = $this->structureManager->getStructure($article->getStructureType(), 'article');
+        $structure->setDocument($article);
+
+        $data = $article->getStructure()->toArray();
+
+        $content = $this->proxyFactory->createProxy(
```
Comment (alexander-schranz) [inline Document/Serializer/ArticleWebsiteSubscriber.php]: make a function call:

```php
return [
    'content' => $this-createContentProxy($structure, $data),
    'view' => $this-createViewProxy($structure, $data),
];
```

File: Resources/doc/article_default.html.twig [hunk 5]
```diff
@@ -2,8 +2,15 @@
 
 {% block content %}
     <h1 property="title">{{ content.title }}</h1>
+    <h2 property="pageTitle">{{ content.pageTitle }}</h2>
 
     <div property="article">
-        {{ content.article|raw }}
+        {{ content.article|default()|raw }}
```
Comment (alexander-schranz) [inline Resources/doc/article_default.html.twig]: why default? this should not be necassery as the main article and the page article has a article field.

File: Routing/ArticleRouteDefaultProvider.php (hunk 1 above)
Comment (alexander-schranz) [inline Routing/ArticleRouteDefaultProvider.php]: add as comment also why (page-number, seo, ...)

File: Resources/doc/article_default.html.twig (hunk 5 above)
Comment (wachterjohannes) [inline Resources/doc/article_default.html.twig]: +1

**Comment by alexander-schranz** [comment]
@wachterjohannes not sure if its part of this pull request but the settings tab is broken.

**Comment by wachterjohannes** [comment]
@alexander-schranz I will merge this and fix the settings tab in an own pull-request

**Pull request merged**